manifests/template/mongo-db.yaml.raw

```diff
@@ -0,0 +1,85 @@
+apiVersion: templating.flanksource.com/v1
+kind: Template
+metadata:
+  name: mongo-db
+spec:
+  source:
+    apiVersion: db.flanksource.com/v1
+    kind: MongoDB
+  resources:
+    - apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: mongodb-{{ .metadata.name }}
+        namespace: '{{ .metadata.namespace }}'
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            app.kubernetes.io/name: mongodb
+            app.kubernetes.io/instance: '{{ .metadata.name }}'
+        strategy:
+          type: Recreate
+        template:
+          metadata:
+            labels:
+              app.kubernetes.io/name: mongodb
+              app.kubernetes.io/instance: '{{ .metadata.name }}'
+          spec:
+            securityContext:
+              fsGroup: 184
+            containers:
+              - image: docker.io/centos/mongodb-36-centos7:1
+                name: mongodb
+                env:
+                  - name: MONGODB_DATABASE
+                    value: "mongo"
+                  - name: MONGODB_USER
+                    value: "mongouser"
+                  - name: MONGODB_PASSWORD
+                    value: "mongouserpassword"
+                  - name: MONGODB_ADMIN_PASSWORD
+                    value: "mongoadminpassword"
```

Comment:
Can we pull these from an autogenerated secret ?

eg.

```
stringData:
      "mongo.password": '{{- kget (print "secret/" .metadata.namespace "/" .metadata.name) "password.password" | default (random.AlphaNum 16) -}}'
```

Reply:
> Can we pull these from an autogenerated secret ?
> 
> eg.
> 
> ```
> stringData:
>       "mongo.password": '{{- kget (print "secret/" .metadata.namespace "/" .metadata.name) "password.password" | default (random.AlphaNum 16) -}}'
> ```

Nice. I was gonna ask if we have a helper like this. But since this is just a stop-gap until we have MongoDB Operator implemented so I just make it "lean" by hard-coding the secret but this is much nicer. I will add that.

Reply:
Done